From 0e490caa6fc533298e06fc89261066d135e42ec4 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 17 Jul 2024 16:33:08 +0200
Subject: [PATCH 670/954] feat(st-clk): move gate_refcounts check to use
 UINT8_MAX

The gate_refcounts check in the _clk_stm32_enable_core function
was using UINT_MAX, which is incorrect. It should use UINT8_MAX instead.
This change fixes the check to use the correct maximum value.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 4a6ca346370731b3a4b4276ef4e2fe39253d3c39 feat(st-clk): move gate_refcounts check to use UINT8_MAX]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I5e6a0789060f95a6b96134a6c00a9c62943865d9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/393870
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/st/clk/clk-stm32-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/st/clk/clk-stm32-core.c b/drivers/st/clk/clk-stm32-core.c
index cfb8fecec4..86a4b71f47 100644
--- a/drivers/st/clk/clk-stm32-core.c
+++ b/drivers/st/clk/clk-stm32-core.c
@@ -472,7 +472,7 @@ static int _clk_stm32_enable_core(struct stm32_clk_priv *priv, int id)
 
 	priv->gate_refcounts[id]++;
 
-	if (priv->gate_refcounts[id] == UINT_MAX) {
+	if (priv->gate_refcounts[id] == UINT8_MAX) {
 		ERROR("%s: %d max enable count !", __func__, id);
 		panic();
 	}
-- 
2.53.0

